week-4/E-arrays-of-objects/exercise-2.js

```diff
@@ -7,50 +7,72 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
 3) Print in the console all the destination names more than 300 kms far away and reachable by train.
 */
 
-
 var destination1 = {
-    destinationName: "Edinburgh",
-    distanceKms: 80,
-    transportations: ["car", "bus", "train"]
+  destinationName: "Edinburgh",
+  distanceKms: 80,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination2 = {
-    destinationName: "London",
-    distanceKms: 650,
-    transportations: ["car", "bus", "train"]
+  destinationName: "London",
+  distanceKms: 650,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination3 = {
-    destinationName: "Paris",
-    distanceKms: 900,
-    transportations: ["train", "plane"]
+  destinationName: "Paris",
+  distanceKms: 900,
+  transportations: ["train", "plane"]
 };
 
 var destination4 = {
-    destinationName: "Dublin",
-    distanceKms: 350,
-    transportations: ["plane", "ferry"]
+  destinationName: "Dublin",
+  distanceKms: 350,
+  transportations: ["plane", "ferry"]
 };
 
-var travelDestinations = [destination1, destination2, destination3, destination4];
+var travelDestinations = [
+  destination1,
+  destination2,
+  destination3,
+  destination4
+];
 
 /* 
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
+var destinationNamesWithin500Kms = travelDestinations
+  .filter(destination => destination.distanceKms < 500)
+  .map(destination => destination.destinationName);
 
+var destinationNameReachableByFerry = travelDestinations
+  .filter(reach => reach.transportations.includes("ferry"))
+  .map(destination => destination.destinationName);
 
```

Comment:
reach isn't a good name for the parameter to the callback function given to filter.  filter will take each destination in turn and check it against the callback function.  Can you think of a better parameter name?

Reply:
destination